redaxo/src/core/lib/console/setup/run.php

```diff
@@ -0,0 +1,367 @@
+            $io->error("Directory permissions error:\n" .implode("\n", $errors));
+            return 1;
+        }
+        $io->success('Directory permissions ok');
+
+        // ---------------------------------- step 4 . Config
+        $io->title('Step 4 of 6 / Creating config');
+
+        $io->section('General');
+        if ($input->getOption('server')) {
+            $config['server'] = $input->getOption('server');
+            $io->success('Using website URL "'.$config['server'].'"');
+        } else {
+            $config['server'] = $io->ask('Website URL', $config['server'], $requiredValue);
+        }
+
+        if ($input->getOption('servername')) {
+            $config['servername'] = $input->getOption('servername');
+            $io->success('Using website name "'.$config['servername'].'"');
+        } else {
+            $config['servername'] = $io->ask('Website name', $config['servername'], $requiredValue);
+        }
+
+        if ($input->getOption('error-email')) {
+            $config['error_email'] = $input->getOption('error-email');
+            $io->success('Using "'.$config['error_email'].'" in case of errors');
+        } else {
+            $config['error_email'] = $io->ask('E-mail address in case of errors', $config['error_email'], $requiredValue);
+        }
+
+        if (!$input->getOption('timezone')) {
+            $q = new Question('Choose timezone', $config['timezone']);
+            $q->setAutocompleterValues(DateTimeZone::listIdentifiers());
+            $q->setValidator(static function ($value) {
+                if (false === @date_default_timezone_set($value)) {
+                    throw new RuntimeException('Time zone invalid');
+                }
+                return $value;
+            });
+            $config['timezone'] = $io->askQuestion($q);
+        } else {
+            $timezone = $input->getOption('timezone');
+            if (!in_array($timezone, DateTimeZone::listIdentifiers(), true)) {
+                throw new InvalidArgumentException('Unknown timezone "'.$timezone.'" specified');
+            }
+            $config['timezone'] = $timezone;
+            $io->success('Timezone "'.$timezone.'" selected');
+        }
+
+        if (!$input->getOption('db-host') && !$input->getOption('db-login') && !$input->getOption('db-password') && !$input->getOption('db-name')) {
+            $io->section('Database information');
+            do {
+                $config['db'][1]['host'] = $io->ask('MySQL host', $config['db'][1]['host']);
+                $config['db'][1]['login'] = $io->ask('Login', $config['db'][1]['login']);
+                $config['db'][1]['password'] = $io->askHidden('Password');
+                $config['db'][1]['name'] = $io->ask('Database name', $config['db'][1]['name']);
+
+                $redaxo_db_create = $io->confirm('Create database', false);
+
+                try {
+                    $err = rex_setup::checkDb($config, $redaxo_db_create);
```

Comment:
Der DB-Check funktioniert an dieser Stelle noch nicht, weil die Config noch nicht vorliegt.

Reply:
fix in #3164